### PR TITLE
Add Security Strengths Appendix

### LINKED ISSUE #71: Security Strengths
Do we need a security strength appendix (between ASN.1 and Examples) like the other I-Ds have? The CMS ML-KEM I-D doesn't have one, but these of them do.

### PR DESCRIPTION
**Pull request by seanturner**
Closes #71.

### CODE REVIEW

File: draft-ietf-lamps-kyber-certificates.md [hunk 1]
```diff
@@ -319,6 +319,30 @@ as per {{RFC5280}}, certificates use the Distinguished Encoding Rules; see
...
+NIST expects to offer notable levels of resistance to both quantum and
+classical attack.  To wit, an algorithm that achieves NIST PQC security
+level 1 must require computational resources to break the relevant
+security property, which are greater than those required for a
+brute-force key search on AES-128.  Levels 3 and 5 use AES-192 and
+AES-256 as reference, respectively.  Levels 2 and 4 use collision search
+for SHA-256 and SHA-384 as reference.
+
+<aside markdown="block">
+  TODO: what should go in this table?
```
Comment (bwesterb) [inline draft-ietf-lamps-kyber-certificates.md:336]: ML-KEM-512, ML-KEM-768, ML-KEM-1024, no?

**Review by bwesterb — APPROVED** [review]

File: draft-ietf-lamps-kyber-certificates.md [hunk 2]
```diff
@@ -319,6 +319,30 @@ as per {{RFC5280}}, certificates use the Distinguished Encoding Rules; see
 <CODE ENDS>
 ~~~
 
+# Security Strengths {#arnold}
+
+Instead of defining the strength of a quantum algorithm in a traditional
+manner using the imprecise notion of bits of security, NIST has instead
+elected to define security levels by picking a reference scheme, which
```
Comment (dconnolly) [inline draft-ietf-lamps-kyber-certificates.md:326]:
```suggestion
manner using the imprecise notion of bits of security, NIST has
defined security levels by picking a reference scheme, which
```

File: draft-ietf-lamps-kyber-certificates.md [hunk 3]
```diff
@@ -319,6 +319,30 @@ as per {{RFC5280}}, certificates use the Distinguished Encoding Rules; see
...
+NIST expects to offer notable levels of resistance to both quantum and
+classical attack.  To wit, an algorithm that achieves NIST PQC security
+level 1 must require computational resources to break the relevant
+security property, which are greater than those required for a
+brute-force key search on AES-128.  Levels 3 and 5 use AES-192 and
+AES-256 as reference, respectively.  Levels 2 and 4 use collision search
+for SHA-256 and SHA-384 as reference.
```
Comment (dconnolly) [inline draft-ietf-lamps-kyber-certificates.md:333]:
```suggestion
classical attack.  To wit, a KEM algorithm that achieves NIST PQC
security must require computational resources to break IND-CCA2
security comparable or greater than that required for key search
on AES-128, AES-192, and AES-256 for Levels 1, 3, and 5, respectively.
Levels 2 and 4 use collision search for SHA-256 and SHA-384 as reference.
```

File: draft-ietf-lamps-kyber-certificates.md [hunk 4]
```diff
@@ -319,6 +319,30 @@ as per {{RFC5280}}, certificates use the Distinguished Encoding Rules; see
...
+<aside markdown="block">
+  TODO: what should go in this table?
+</aside>
+
+| Level | ? | Encap. Key | Decap. Key | Ciphertext | SS |
+|-      |-  |-           |-           |-           |-   |
+| 1     |   | 800        | 1632       | 768        | 32 |
+| 3     |   | 1184       | 2400       | 1952       | 32 |
+| 5     |   | 1568       | 3168       | 2592       | 32 |
+{: #tab-strengths title="ML-KEM security strengths"}
```
Comment (dconnolly) [inline draft-ietf-lamps-kyber-certificates.md:344]:
```suggestion
| Level | Parameter Set | Encap. Key | Decap. Key | Ciphertext | SS |
|-      |-              |-           |-           |-           |-   |
| 1     | ML-KEM-512    | 800        | 1632       | 768        | 32 |
| 3     | ML-KEM-768    | 1184       | 2400       | 1952       | 32 |
| 5     | ML-KEM-1024   | 1568       | 3168       | 2592       | 32 |
{: #tab-strengths title="ML-KEM security strengths"}
```

File: draft-ietf-lamps-kyber-certificates.md (hunk 1 above)
Comment (seanturner) [reply draft-ietf-lamps-kyber-certificates.md:336]: This is obviously the right thing to include. I was just thinking that maybbe we should include references to 𝑛 𝑞 𝑘 𝜂1 𝜂2 𝑑𝑢 𝑑𝑣 ;)

File: draft-ietf-lamps-kyber-certificates.md (hunk 3 above)
Comment (bwesterb) [reply draft-ietf-lamps-kyber-certificates.md]: sgtm. Nit: it's IND-CCA (without the 2) and I think it's better to put it in parentheses so:

>  (...) require computational resources to break (IND-CCA) security comparable (...)

File: draft-ietf-lamps-kyber-certificates.md (hunk 1 above)
Comment (bwesterb) [reply draft-ietf-lamps-kyber-certificates.md:335]: Not sure how these other parameters help the reader.

**Review by bwesterb — APPROVED** [review]

**Pull request merged**